runtime/server/x86/bin/decoder_main.cc

```diff
@@ -1,54 +1,88 @@
+#include <chrono>
+#include <iomanip>
+#include <string>
 
 #include "decoder/symbol_table.h"
 #include "decoder/torch_asr_decoder.h"
 #include "decoder/torch_asr_model.h"
 #include "frontend/feature_pipeline.h"
 #include "frontend/wav.h"
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+#include "torch/script.h"
+#include "torch/torch.h"
 
 DEFINE_int32(num_bins, 80, "num mel bins for fbank feature");
 DEFINE_int32(chunk_size, 16, "decoding chunk size");
+DEFINE_int32(num_threads, 1, "num threads for device");
 DEFINE_string(model_path, "", "pytorch exported model path");
-DEFINE_string(wav_path, "", "wav path");
+DEFINE_string(wav_scp, "", "input wav scp");
 DEFINE_string(dict_path, "", "dict path");
+DEFINE_string(result, "", "result output file");
 
 int main(int argc, char *argv[]) {
   google::ParseCommandLineFlags(&argc, &argv, false);
   google::InitGoogleLogging(argv[0]);
 
-  wenet::WavReader wav_reader(FLAGS_wav_path);
-  wenet::FeaturePipelineConfig feature_config;
-  feature_config.num_bins = FLAGS_num_bins;
-  auto feature_pipeline =
-      std::make_shared<wenet::FeaturePipeline>(feature_config);
-  feature_pipeline->AcceptWaveform(std::vector<float>(
-      wav_reader.data(), wav_reader.data() + wav_reader.num_sample()));
-  feature_pipeline->set_input_finished();
-  LOG(INFO) << "num frames " << feature_pipeline->num_frames();
-
   auto model = std::make_shared<wenet::TorchAsrModel>();
-  model->Read(FLAGS_model_path);
-
+  model->Read(FLAGS_model_path, FLAGS_num_threads);
   wenet::SymbolTable symbol_table(FLAGS_dict_path);
   wenet::DecodeOptions decode_config;
   decode_config.chunk_size = FLAGS_chunk_size;
-  wenet::TorchAsrDecoder decoder(feature_pipeline, model, symbol_table,
-                                 decode_config);
-
-  while (true) {
-    bool finish = decoder.Decode();
-    if (finish) {
-      LOG(INFO) << "Final result: " << decoder.result();
-      break;
-    } else {
+  wenet::FeaturePipelineConfig feature_config;
+  feature_config.num_bins = FLAGS_num_bins;
+
+  std::ofstream result;
+  result.open(FLAGS_result);
+
+  std::ifstream wav_scp(FLAGS_wav_scp);
+  std::string line;
+  unsigned int total_waves_dur = 0;
+  unsigned int total_decode_time = 0;
+  while (getline(wav_scp, line)) {
+    std::string delimiter = " ";
+    int pos = line.find(delimiter);
+    std::string utt = line.substr(0, pos);
+    std::string wav = line.substr(pos + delimiter.length(), line.length());
+
+    wenet::WavReader wav_reader(wav);
+    auto feature_pipeline =
+        std::make_shared<wenet::FeaturePipeline>(feature_config);
+    feature_pipeline->AcceptWaveform(std::vector<float>(
+        wav_reader.data(), wav_reader.data() + wav_reader.num_sample()));
+    feature_pipeline->set_input_finished();
+    LOG(INFO) << "num frames " << feature_pipeline->num_frames();
+
+    wenet::TorchAsrDecoder decoder(feature_pipeline, model, symbol_table,
+                                   decode_config);
+
```

Comment:
add check_eq(wav_reader.sample_rate(), 16000)

add it's better be:
wave_dur = wav_reader.num_sample() / 16000 * 1000